master/buildbot/reporters/telegram.py

```diff
@@ -0,0 +1,1055 @@
+                for e in evs
+            ]
+            for evs in (('started', 'finished'), ('success', 'failure'), ('warnings', 'exception'),
+                        ('problem', 'recovery'), ('worse', 'better'), ('cancelled', 'worker'))
+        ] + [[self.query_button("Hide...", '/notify list')]]
+
+        if tquery:
+            self.bot.edit_keyboard(self.chatid, tquery['message']['message_id'], keyboard)
+        else:
+            self.send("Here are available notifications and their current state. "
+                      "Click to turn them on/off.",
+                      reply_markup={'inline_keyboard': keyboard})
+
+    def ask_for_reply(self, prompt, greeting='Ok'):
+        kwargs = {}
+        if self.userid != self.chatid:
+            username = self.user.get('username', '')
+            if username:
+                if greeting:
+                    prompt = "{} @{}, now {}...".format(greeting, username, prompt)
+                else:
+                    prompt = "@{}, now {}...".format(username, prompt)
+                kwargs['reply_markup'] = {
+                    'force_reply': True,
+                    'selective': True
+                }
+            else:
+                if greeting:
+                    prompt = "{}, now reply to this message and {}...".format(greeting, prompt)
+                else:
+                    prompt = "Reply to this message and {}...".format(prompt)
+        else:
+            if greeting:
+                prompt = "{}, now {}...".format(greeting, prompt)
+            else:
+                prompt = prompt[0].upper() + prompt[1:] + "..."
+            kwargs['reply_markup'] = {
+                'force_reply': True
+            }
+        self.send(prompt, **kwargs)
+
+    @defer.inlineCallbacks
+    @Contact.overrideCommand
+    def command_STOP(self, args, **kwargs):
+        argv = self.splitArgs(args)
+        if len(argv) >= 3 or \
+                argv and argv[0] != 'build':
+            super().command_STOP(args)
+            return
+        argv = argv[1:]
+        if not argv:
+            builders = yield self.get_running_builders()
+            if builders:
+                keyboard = [
+                    [self.query_button("🚫 " + b, '/stop build {}'.format(b))]
+                    for b in builders
+                ]
+                self.send("Select builder to stop...",
+                          reply_markup={'inline_keyboard': keyboard})
+        else:  # len(argv) == 1
+            self.template = '/stop ' + args + ' {}'
+            self.ask_for_reply("give me the reason to stop build on `{}`".format(argv[0]))
+
+    @Contact.overrideCommand
+    def command_SHUTDOWN(self, args, **kwargs):
+        if args:
+            return super().command_SHUTDOWN(args)
+        shuttingDown = self.master.botmaster.shuttingDown
+        keyboard = [[
+             self.query_button("🔙 Stop Shutdown", '/shutdown stop')
+             if shuttingDown else
```

Comment:
I think this is not intuitive formatting. Could we move the conditional out of the list definition?

Reply:
Ok, done.